docs/src/main/asciidoc/se/grpc/server.adoc

```diff
@@ -143,25 +146,112 @@ The service implementation will be very similar to our original implementation:
 include::{sourcedir}/se/grpc/ServerSnippets.java[tag=snippet_2, indent=0]
 ----
 
-<1> Specify the proto descriptor in order to provide necessary type information and
+<1> Specify the proto descriptor in order to provide the necessary type information and
 enable Protobuf marshalling.
-<2> Define unary method `Echo` and map it to the `this::echo` handler.
+<2> Define the unary method `Echo` and map it to the `this::echo` handler.
 <3> Create a handler for the `Echo` method, using Protobuf message types for request and response.
-<4> Extract message string from the request.
-<5> Create the response containing extracted message.
-<6> Send the response back to the client by completing response observer.
+<4> Extract the message string from the request.
+<5> Create the response containing the extracted message.
+<6> Send the response back to the client by completing the response observer.
 
 NOTE: The `complete` method shown in the example above is just one of many helper
 methods available in the `ResponseHelper` class. See the full list
 link:{grpc-server-javadoc-base-url}/io/helidon/webserver/grpc/ResponseHelper.html[here].
 
+=== Server Interceptors
+
+gRPC supports the notion of _server interceptors_ which are useful to implement
+common tasks (cross-cutting concerns) on any subset of methods and services.
+Interceptors implement the `io.grpc.ServerInterceptor` interface which defines
+a single `interceptCall` method. Server interceptors are arranged in a _chain_ that
+wraps the actual gRPC method invocation and that can be ordered by weight.
+
+Server interceptors are registered while creating routes, and will intercept any
+gRPC method calls that follow: that is, registration ordering is important to
+make sure interceptor chains are properly constructed to accomplish a
+certain task. For example, consider the following routing definition:
+
+[source,java]
+----
+include::{sourcedir}/se/grpc/ServerSnippets.java[tag=snippet_4, indent=0]
+----
+
+This routing includes two server interceptor instances of types `Interceptor1` and
+`Interceptor2`. As stated above, the order in which these interceptors are
+registered is important. In this example, `Interceptor1` will be called for
+any method in `EchoService` and `MathService`, but not for `GreetService`.
+Similarly, `Interceptor2` will be called only for methods in `MathService`.
+The default weight of a server interceptor is `InterceptorWeights.USER`;
+it follows that for `MathService`, `Interceptor2` will be called _before_
+`Interceptor1` given its _higher_ weight of `Interceptor.USER + 100`.
+
+NOTE: Even though the gRPC API supports interception of methods via
+alternative mechanisms, it is recommended to use the `intercept` method
+on a `GrpcRouting` builder, as shown above, to ensure correct ordering
+based on weights.
+
+=== Metrics
+
```

Comment:
Same question as earlier about maybe mentioning adding a dep. on `helidon-metrics-providers-micrometer` (or `helidon-metrics` which also includes the built-in system meters)?

Reply:
See above